test/org/pcic/util/MockSteps.groovy

```diff
@@ -0,0 +1,20 @@
+package util
+
+
+class MockSteps implements Serializable {
+    Map env
+
+    MockSteps(steps) {
+        this.env = [PCIC_DOCKER_DEV01: 'used-for-testing']
+    }
+
+    /**
+     * Mocks the behaviour of `sh`
+     */
+    def sh(Map args) {
+        // Return some tags
+        if(args.script == 'git tag --contains' && args.containsKey('returnStdout')) {
+            return "1.0.0\nsome-tag"
+        }
```

Comment:
What does it (or should it) return if the condition is false? What happens in this language when you don't have an explicit `return` statement?

Reply:
That particular condition is when the `returnStdout` flag is set to `true`.  In that case some string is expected to be returned.  Otherwise the `sh` command doesn't return anything.  In groovy if have something like this:
```groovy
def myMethod(val) {
    if (val > 1) {
        return 'some-string'
    }
}

def test = myMethod(0)
```
The class of `test` will be a `NullObject`.  You can still call `m1()` without assigning the output a  to a variable and it won't complain either way (return value, or no return value).